extensions/amp-analytics/0.1/test/test-linker-manager.js

```diff
@@ -628,6 +628,30 @@ describes.realWin('Linker Manager', {amp: true}, env => {
       });
     });
 
+    it('should not rewrite url if href is fragment', () => {
+      const lm = new LinkerManager(ampdoc, config, /* type */ null, element);
+      return lm.init().then(() => {
+        // Using a real anchor el here because of a.href getter will actually
+        // return the full url, not just the fragment.
+        const a = doc.createElement('a');
+        a.href = '#hello';
+        a.hostname = 'amp.source.com';
+
+        const setterSpy = sandbox.spy();
+        const aProxy = new Proxy(a, {
+          get: (obj, prop) => {
+            return obj[prop];
+          },
+          set: setterSpy,
+        });
+
+        anchorClickHandlers.forEach(handler =>
+          handler(aProxy, {type: 'click'})
+        );
+        expect(setterSpy.notCalled).to.be.true;
```

Comment:
drive-by test convention nit: 🚗
```js
expect(set).not.to.have.been.called;
```
This will give you more specific error messages since it knows it's a function. You can go further to alias it:

```js
expect(set, 'set on a[href=#fragment]').not.to.have.been.called;
```